docs/docs/tutorial/part-4/index.mdx

```diff
@@ -0,0 +1,918 @@
+
+- Use **GraphiQL** to explore the data in the data layer and build your own GraphQL queries.
+- Use the **`useStaticQuery`** hook to pull data into a "building-block" component.
+- Use the **`gatsby-source-filesystem`** plugin to pull data into your site from your computer's filesystem.
+- Create a **page query** to pull data into a page component.
+
+## Meet Gatsby's GraphQL data layer
+
+Gatsby has its own [GraphQL](https://graphql.org/) data layer where it keeps all the data for your site. But how does it work under the hood?
+
+First, your data is stored in one or more **sources**. That source might be a folder on your computer's filesystem, a content management system (CMS) like WordPress, or a database. You might even have multiple data sources!
+
+How do you get from its source into the data layer? By adding special plugins to your site called **source plugins**. Each source plugin is designed to communicate with a specific source. When you build your site, each source plugin pulls data from its particular source and adds it to the GraphQL data layer for your site.
+
+<Announcement style={{marginBottom: "1.5rem"}}>
+
+**Tip:** Curious what source plugins are in the [Plugin Library](/plugins/)? You can identify source plugins by their name: they typically start with `gatsby-source-`.
+
+For example, a few popular source plugins are `gatsby-source-filesystem` and `gatsby-source-contentful`.
+
+</Announcement>
+
+How do you get data back out of the data layer? You can write **GraphQL queries** inside of your components to pull out the data you want to use in your site. When you build your site, Gatsby will find all the GraphQL queries in your components, run them, and pass the resulting data into your component as a `data` prop.
+
+
+![A diagram showing how data flows into and out of the GraphQL data layer. Source plugins pull data out of a particular data source and into the data layer for your site. GraphQL queries to pull data out of the data layer and into your React components.](./data-layer.png)
+
+
+## Use GraphiQL to explore the data layer and write GraphQL queries
+
+How do you know what data is in your site's GraphQL data layer? When you start the local development server for your site, Gatsby automatically creates a special endpoint that lets you use an in-browser tool called **GraphiQL**. With GraphiQL, you can explore your site's data and build GraphQL queries.
+
+Follow the steps below to open the GraphiQL interface:
+
+1. Start up your local development server by running `gatsby develop`.
+1. In a web browser, go to `http://localhost:8000/___graphql`. (That's three underscores in the URL.)
+
+![A screenshot of the GraphiQL interface in a web browser. It has three main sections: the Explorer, the Query Editor, and the Result Window.](./graphiql.png)
+
+<Collapsible
+  summary={<h3>A closer look at the GraphiQL interface</h3>}
+>
+
+There are three main sections of the GraphiQL interface:
```

Comment:
As a cherry on top maybe also explain Prettify, History, and Code Exporter?

Reply:
Oh neat I've never actually clicked any of those buttons! 😂 I'll add a clean-up story to add it after this PR, since I'll have to do some new writing.